File: payroll-api/src/main/java/com/company/payroll/service/StaffSalaryService.java
package com.company.payroll.service;

import java.util.Optional;

import com.company.payroll.model.StaffSalary;
import com.github.pagehelper.PageInfo;

public interface StaffSalaryService {
	
	/**
	 * Get StaffSalary object by salary id
	 * @param sId salary id
	 * @return {@link StaffSalary} object
	 */
	Optional<StaffSalary> findById(Integer sId);
	
	/**
	 * Method of listing StaffSalary
	 * @param page page number
	 * @param offset page data limit
	 * @return List of StaffSalary
	 */
	PageInfo<StaffSalary> list(int page, int offset);
	
	/**
	 * Method of update StaffSalary
	 * @param staffSalary {@link StaffSalary} object
	 * @return Updated {@link StaffSalary} object
	 */
	StaffSalary update(StaffSalary staffSalary);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/service/StaffSalaryService.kt
package com.company.payroll.service

import com.company.payroll.model.StaffSalary

import com.github.pagehelper.PageInfo

interface StaffSalaryService {
  /**
   *
   * @param sid
   * @return
   */
  fun findById(sid: Int): StaffSalary

  /**
   *
   * @return
   */
  fun list(page: Int, offset: Int): PageInfo<StaffSalary>

  /**
   *
   * @param staffSalary
   * @return
   */
  fun update(staffSalary: StaffSalary): StaffSalary
}